core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -906,6 +939,81 @@ private CompiledQuery compile0(@NotNull CharSequence query, @NotNull SqlExecutio
         return executor.execute(executionContext);
     }
 
+    /*
+     * Allows processing of batches of sql statements (sql scripts) separated by ';' .
+     * Each query is processed in sequence and processing stops on first error and whole batch gets discarded .
+     * Noteworthy difference between this and 'normal' query is that all empty queries get ignored, e.g.
+     * <br>
+     * select 1;<br>
+     * ; ;/* comment \*\/;--comment\n; - these get ignored <br>
+     * update a set b=c  ; <br>
+     * <p>
+     * Useful PG doc link :
+     *
+     * @param query            - block of queries to process
+     * @param batchCallback    - callback to perform actions prior to or after batch part compilation, e.g. clear caches or execute command
+     * @see <a href="https://www.postgresql.org/docs/current/protocol-flow.html#id-1.10.5.7.4">PostgreSQL documentation</a>
+     */
+    public void compileBatch(@NotNull CharSequence query, @NotNull SqlExecutionContext executionContext, BatchCallback batchCallback)
+            throws SqlException, PeerIsSlowToReadException, PeerDisconnectedException {
+        clear();
+        lexer.of(query);
+        isSingleQueryMode = false;
+
+        if (batchCallback == null) {
+            batchCallback = EMPTY_CALLBACK;
+        }
+
+        int position;
+
+        while (lexer.hasNext()) {
+            //skip over empty statements that'd cause error in parser
+            position = getNextValidTokenPosition();
+            if (position == -1) {
+                return;
+            }
+
+            batchCallback.preCompile(this);
+            clear();//we don't use normal compile here because we can't reset existing lexer
+            CompiledQuery current = compileInner(executionContext);
+            //We've to move lexer because some query handlers don't consume all tokens (e.g. SET )
+            //some code in postCompile might need full text of current query
+            CharSequence currentQuery = query.subSequence(position, goToQueryEnd());
+            batchCallback.postCompile(this, current, currentQuery);
+        }
+    }
+
+    private int getNextValidTokenPosition(){
+        while (lexer.hasNext()){
+            CharSequence token = SqlUtil.fetchNext(lexer);
+            if ( token == null ){
+                return -1;
+            } else if (!isSemicolon(token)) {
+                lexer.unparse();
+                return lexer.lastTokenPosition();
+            }
+        }
+
+        return -1;
+    }
+
+    private  int goToQueryEnd() {
+        CharSequence token;
+        lexer.unparse();
+        while ( lexer.hasNext() ){
+            token = SqlUtil.fetchNext(lexer);
+            if (token == null || isSemicolon(token)) {
+                break;
+            }
+        }
+
+        return lexer.getPosition();
+    }
+
+    private boolean isSemicolon(CharSequence token) {
```

Comment:
This method could easily be part of `SqlKeywords`, not a strong feeling regarding this... `;` is not a keyword but it sort of is.